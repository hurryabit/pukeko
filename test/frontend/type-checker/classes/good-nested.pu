class C a where
  c :: a -> a
f :: (C a) => a -> a
f = let g = c in g
