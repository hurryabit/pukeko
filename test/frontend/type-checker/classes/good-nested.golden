data C a =
       | .C (a -> a)
c : forall a. C a -> a -> a =
  fun @a (dict : C a) ->
    match dict with
    | .C c -> c
f : forall a. C a -> a -> a =
  fun @a (c.a : C a) ->
    let g : forall _1. C _1 -> _1 -> _1 =
          fun @_1 (c._1 : C _1) -> c @_1 c._1
    in
    g @a c.a
